src/streamlink/plugins/mediavitrina.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
+from streamlink.utils.url import update_qsd
+
+log = logging.getLogger(__name__)
+
+
+class MediaVitrina(Plugin):
+    _re_url_1 = re.compile(r'https?://(?P<channel>ctc(?:love)?|chetv|domashniy|5-tv)\.ru/(?:online|live)')
+    _re_url_2 = re.compile(r'https?://(?P<channel>ren)\.tv/live')
+    _re_url_3 = re.compile(r'https?://player\.mediavitrina\.ru/(?P<channel>[^/?]+.)(?:/[^/]+)?/[\w_]+/player\.html')
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return (
+            cls._re_url_1.match(url) is not None
+            or cls._re_url_2.match(url) is not None
+            or cls._re_url_3.match(url) is not None
+        )
+
+    def _get_streams(self):
+        channel = (self._re_url_1.match(self.url)
+                   or self._re_url_2.match(self.url)
+                   or self._re_url_3.match(self.url)
+                   ).group("channel")
+
+        if channel in ("5-tv", "tv-5", "5tv"):
+            path, channel = ("tv5", "tv-5")
+        elif channel in ("chetv", "ctc-che", "che_ext"):
+            path, channel = ("ctc", "ctc-che")
+        elif channel == "ctc":
+            path, channel = ("ctc", "ctc")
+        elif channel in ("ctclove", "ctc-love", "ctc_love_ext"):
+            path, channel = ("ctc", "ctc-love")
+        elif channel in ("domashniy", "ctc-dom", "domashniy_ext"):
+            path, channel = ("ctc", "ctc-dom")
+        elif channel == "iz":
+            path, channel = ("iz", "iz")
+        elif channel == "mir":
+            path, channel = ("mtrkmir", "mir")
+        elif channel == "muztv":
+            path, channel = ("muztv", "muztv")
+        elif channel in ("ren", "ren-tv", "rentv"):
+            path, channel = ("nmg", "ren-tv")
+        elif channel == "russia1":
+            path, channel = ("vgtrk", "russia1")
+        elif channel == "russia24":
+            path, channel = ("vgtrk", "russia24")
+        elif channel in ("russiak", "kultura"):
+            path, channel = ("vgtrk", "russiak")
+        elif channel == "spas":
+            path, channel = ("spas", "spas")
+        elif channel == "tvc":
+            path, channel = ("tvc", "tvc")
+        elif channel in ("tvzvezda", "zvezda"):
+            path, channel = ("zvezda", "zvezda")
+        elif channel in ("u", "u_ott"):
+            path, channel = ("utv", "u_ott")
+        else:
+            log.error(f"Unsupported channel: {channel}")
+            return
```

Comment:
It would be better if you'd add a dictionary here that maps the channel input names to path+channel properties.
This makes it much more legible and easier to maintain. And you avoid having lots of unnecessary code branches.